### DISCUSSION TITLE
single-spa is loaded on the page multiple times using module federation

### DISCUSSION THREAD

**Discussion by glats** [module-federation/module-federation-examples#1306] [Q&A] [answered]
I am getting into single-spa with module federation. I created this https://github.com/j1cs/wp5-mf-single-spa but I have two errors which I don't know how to solve:
```
single-spa.min.js:2 single-spa minified message #41: See https://single-spa.js.org/error/?code=41
single-spa.min.js:2 single-spa minified message #1: See https://single-spa.js.org/error/?code=1
``` 
**41: single-spa is loaded on the page multiple times**  
**1: singleSpa.start() was not called**

In this file https://github.com/j1cs/wp5-mf-single-spa/blob/master/packages/navigation/src/NavBar.tsx#L4 I am using navigateToUrl from the single-spa library. When removing this import the application doesn't show any warning.  
In the end i don't know how to handle the share of the dependency.

**Comment by glats** [comment, marked as answer]
I got support from blue collar coder Jack in particular and he recommended me to restructure the project. Which I did and now it works.

**Reply by ScriptedAlchemy** [reply]
Open a pr with an example here so others can benefit. I'll also get docs updated on single spa

**Reply by glats** [reply]
sincerely talking it was something dumb. i've just tell to module federation my deps. in this case single-spa and single-spa-react
